agent/agent_endpoint.go

```diff
@@ -45,7 +45,11 @@ type Self struct {
 
 type XDSSelf struct {
 	SupportedProxies map[string][]string
-	Port             int
+	// Port could be used for either TLS or plain-text communication
+	// up through version 1.14. In order to maintain backwards-compatibility,
+	// Port will now default to TLS and fallback to the standard port value.
+	Port    int
```

Comment:
Note that the semantics of this `Port` field are changing somewhat. Rather than being a direct representation of the `ports.grpc` field, it is instead referring to the "preferred" port so that backwards compatibility is not broken.

If you look at the first commit in this PR, you will see that an additional `PortPlainText` field was added prior to making backwards compatibility tweaks. This should likely be re-introduced after the deprecated TLS compatibility mode is eventually removed.

Reply:
This documentation is why I was hesitant to change it that way.

https://www.consul.io/api-docs/agent#read-configuration

"its format will not change in a backwards incompatible way between releases."

Reply:
Actually, I just realized that I accidentally thought the `XDS` field was nested underneath the `Config` element, but it's not. So I suppose it's safe to change (I doubt anyone would reference XDS but us).